runtime/vam/expr/gencomparefuncs.go

```diff
@@ -0,0 +1,316 @@
+//go:build ignore
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/brimdata/zed/vector"
+)
+
+func main() {
+	var buf bytes.Buffer
+	fmt.Fprintln(&buf, "// Code generated by gencomparefuncs.go. DO NOT EDIT.")
+	fmt.Fprintln(&buf)
+	fmt.Fprintln(&buf, "package expr")
+	fmt.Fprintln(&buf, "import (")
+	fmt.Fprintln(&buf, `"bytes"`)
+	fmt.Fprintln(&buf, `"strings"`)
+	fmt.Fprintln(&buf) // Put subsequent imports in a new section.
+	fmt.Fprintln(&buf, `"github.com/brimdata/zed/vector"`)
+	fmt.Fprintln(&buf, ")")
+
+	var ents strings.Builder
+	for _, op := range []string{"==", "!=", "<", "<=", ">", ">="} {
+		for _, typ := range []string{"Int", "Uint", "Float", "String", "Bytes"} {
+			for lform := vector.Form(0); lform < 3; lform++ {
+				for rform := vector.Form(0); rform < 3; rform++ {
+					if lform == vector.FormConst && rform == vector.FormConst {
+						// no const x const
+						continue
+					}
+					fmt.Fprintln(&buf, genFunc(op, typ, lform, rform))
+					fmt.Fprintf(&ents, "%d: %s,\n", vector.CompareOpCode(op, vector.KindFromString(typ), lform, rform), funcName(op, typ, lform, rform))
+				}
+			}
+		}
+	}
+
+	fmt.Fprintln(&buf, "var compareFuncs = map[int]func(vector.Any, vector.Any) *vector.Bool{")
+	fmt.Fprintln(&buf, ents.String())
+	fmt.Fprintln(&buf, "}")
+
+	out, formatErr := format.Source(buf.Bytes())
+	if formatErr != nil {
+		// Write unformatted source so we can find the error.
+		out = buf.Bytes()
+	}
+	const fileName = "comparefuncs.go"
+	if err := os.WriteFile(fileName, out, 0644); err != nil {
+		log.Fatal(err)
+	}
+	if formatErr != nil {
+		log.Fatal(fileName, ":", formatErr)
+	}
+}
+
+func genFunc(op, typ string, lhs, rhs vector.Form) string {
+	var s string
+	s += "func "
+	s += funcName(op, typ, lhs, rhs)
+	s += "(lhs, rhs vector.Any) *vector.Bool {\n"
+	s += "n := lhs.Len()\n"
+	s += "out := vector.NewBoolEmpty(n, nil)\n"
+	s += genVarInit("l", typ, lhs)
+	s += genVarInit("r", typ, rhs)
+	s += formToLoop(typ, lhs, rhs, op)
+	s += "return out\n"
+	s += "}\n"
+	return s
+}
+
+func genVarInit(which, typ string, form vector.Form) string {
+	switch form {
+	case vector.FormFlat:
+		return fmt.Sprintf("%s := %shs.(*vector.%s)\n", which, which, typ)
+	case vector.FormDict:
+		s := fmt.Sprintf("%sd := %shs.(*vector.Dict)\n", which, which)
+		s += fmt.Sprintf("%s := %sd.Any.(*vector.%s)\n", which, which, typ)
+		s += fmt.Sprintf("%sx := %sd.Index\n", which, which)
+		return s
+	case vector.FormConst:
+		return fmt.Sprintf("%sconst, _ := %shs.(*vector.Const).As%s()\n", which, which, typ)
+	default:
+		panic("genVarInit: bad form")
+	}
+}
+
+const flat_flat = `
+	for k := uint32(0); k < n; k++ {
+		if l.Values[k] %s r.Values[k] {
+			out.Set(k)
+		}
+	}
+`
```

Comment:
Maybe get rid of the extra line here so the generated file doesn't have the extra line?

Reply:
Done in 8135e285738f3be5e950ecd2ef8162904d4659a6.